### PR TITLE
docs: prepare Falcon `4.0.0a1`

### PR DESCRIPTION
**Pull request by vytas7**

### CODE REVIEW

**Review by CaselIT — DISMISSED** [review]
docs look fine. left a suggestion

File: docs/changes/4.0.0.rst [hunk 1]
```diff
@@ -32,17 +45,37 @@ now typed, further type annotations may be added throughout the 4.x release cycl
...
-  All type aliases in falcon are considered private, and if used should be
-  imported inside ``if TYPE_CHECKING:`` blocks to avoid possible import errors
-  after an update.
+        class MyResource:
+            def on_get(self, req: falcon.Request, resp: falcon.Response) -> None:
+                resp.media = {'message': 'Hello, World!'}
+
+    If you do decide to reuse these aliases for some reason anyway, they should
+    preferably be imported inside  :python:`if TYPE_CHECKING:` blocks in order
+    to avoid possible runtime errors after an update.
```
[the remark this answers is not in the document]
Comment (CaselIT) [reply docs/changes/4.0.0.rst:65]: maybe also mention the public `falcon.typing` module?

[the remark this answers is not in the document]
Comment (vytas7) [reply docs/changes/4.0.0.rst:65]: Aha, didn't notice we actually have one :slightly_smiling_face: 

Yes, I'll write a note on that. Maybe we should add it as a docs section then?

[the remark this answers is not in the document]
Comment (CaselIT) [reply docs/changes/4.0.0.rst:65]: we will have one :).

Where do you think it's appropriate to add it?

[the remark this answers is not in the document]
Comment (vytas7) [reply docs/changes/4.0.0.rst:65]: (Or "Type Aliases", "Type Annotations", not sure what is the best name for this rubric.)

[the remark this answers is not in the document]
Comment (CaselIT) [reply docs/changes/4.0.0.rst:65]: We don't really have much. Maybe a section inside utilities is better for now?

[the remark this answers is not in the document]
Comment (vytas7) [reply docs/changes/4.0.0.rst:65]: Makes sense, I didn't know exactly how much we were planning to write.
If it is just 1-2 small paragraphs, we can place it under utilities too.

**Review by CaselIT — APPROVED** [review]

**Pull request merged**